File: sdk/python/examples/user_input_roundtrip.py
from __future__ import annotations

import os

from openyak_sdk import OpenyakClient


def main() -> None:
    with OpenyakClient(base_url=os.environ["OPENYAK_BASE_URL"]) as client:
        thread = client.create_thread(
            model="claude-sonnet-4-6",
            allowed_tools=["read_file"],
        )
        paused = thread.run("PARITY_SCENARIO:request_user_input_roundtrip")
        if paused.status != "awaiting_user_input":
            raise RuntimeError(f"expected awaiting_user_input, got {paused.status}")

        resumed = thread.resume_user_input(
            request_id=paused.pending_user_input.request_id,
            content="feature",
            selected_option="feature",
        )
        print(resumed.status)
        print(resumed.final_text)


if __name__ == "__main__":
    main()

File: sdk/typescript/examples/user-input-roundtrip.ts
import { OpenyakClient } from "../src/index.js";

const baseUrl = process.env.OPENYAK_BASE_URL;
if (!baseUrl) {
  throw new Error("Set OPENYAK_BASE_URL to a running local `openyak server` address.");
}
const operatorToken = process.env.OPENYAK_OPERATOR_TOKEN;

const client = new OpenyakClient({
  baseUrl,
  ...(operatorToken === undefined ? {} : { operatorToken }),
});
const thread = await client.createThread({
  model: "claude-sonnet-4-6",
  allowedTools: ["read_file"],
});

const paused = await thread.run("PARITY_SCENARIO:request_user_input_roundtrip");
if (paused.status !== "awaiting_user_input") {
  throw new Error(`Expected awaiting_user_input, received ${paused.status}`);
}

const resumed = await thread.resumeUserInput({
  requestId: paused.pendingUserInput.request_id,
  content: "feature",
  selectedOption: "feature",
});

console.log({
  paused,
  resumed,
});
